apache/couchdb-fauxton
# Refactor map/reduce views to use Redux

## Overview


- [x] Updated map/reduce views to use Redux version of IndexResults components
- [x] Removed non-Redux components from `index-results` 
- [x] Removed Mango resources from `documents/resources.js` 
- [x] Query Options are now preserved when switching between different views, all docs, design docs. Also, if any of the options is enabled (except for All docs) the header is highlighted to indicate this fact.

Extra 
- [x] Bug fix: when creating a new view in an existing design doc, an error message is showed and the page doesn't change to the new view.
- [x] Bug fix: handle error when saving view. Closes #914 

## Testing recommendations

- Select a view and browse the results
  - Change display mode (Table/Metadata/JSON)
  - Click on next/previous page
  - Change the num of 
  - Set query options and click Run Query
    - Check that {}JSON link includes options from Query Options
    - Check that Reduce option only shows for views that contain a reduce function
       - When the Reduce option is enabled, only the Metadata layout is enabled
 - Create a new view in an existing design doc

## Checklist
